### PR TITLE
add build trigger docs

### PR DESCRIPTION
**Pull request by bparees**
Still need to discuss a couple details about registry/repos and webhooks with some people before finishing this.

### CODE REVIEW

**Comment by bparees** [comment]
@csrwng @soltysh review please.

**Comment by csrwng** [comment]
@bparees - I think we should merge this with the existing webhook doc: https://github.com/openshift/openshift-docs/blob/master/using_openshift/webhooks.adoc 
Or is the intent to replace that doc?

**Comment by bparees** [comment]
@csrwng doh, yes it should be merged with that.  not sure how i missed it being there, anyway i'll fold the content together and update my pull.

**Comment by bparees** [comment]
@csrwng updated.

**Comment by csrwng** [comment]
You need to update the index at _build_cfg.yml so your doc can appear instead of the existing webhooks doc.

**Comment by bparees** [comment]
done, thanks

**Comment by nhr** [comment]
@bparees please rebase and push again; when that's done I will merge.

**Comment by bparees** [comment]
@nhr rebased.

**Comment by nhr** [comment]
Thanks, merging.

File: using_openshift/builds.adoc [hunk 1]
```diff
@@ -37,3 +37,115 @@ Build can be manually canceled using following command:
...
+* Image change
+
+=== Webhook triggers
+Webhook triggers allow you to trigger a new build by sending a request to the OpenShift API endpoint.  You can define two types of webhook triggers:  
+
+* https://developer.github.com/webhooks/[Github Webhooks]
+* Generic Webhooks
+
+==== Github webhooks
+Github webhooks can handle the call made by Github when a repository is updated.  When defining the trigger you can specify a `secret` which you will supply to Github when configuring the webhook.  This will ensure that only you and your repository can trigger the build.  The trigger definition json within the BuildConfig looks like this:
```
Comment (csrwng) [inline using_openshift/builds.adoc]: > When defining the trigger you can specify a `secret` which you will supply to Github when configuring the webhook.
I think it'd be more accurate to say that the secret will be part of the URL that you supply to Github

Comment (bparees) [inline using_openshift/builds.adoc]: alas @nhr beat you to the punch.  i'll edit this in a followup.

Comment (csrwng) [inline using_openshift/builds.adoc]: yup, I think it's minor. The change would need to be made for both github and generic.

Comment (bparees) [inline using_openshift/builds.adoc]: I think the generic hook wording is probably ok as is.  you do have to provide the secret when making the request (in the url).  close enough imho.

**Pull request merged**